File: app/src/main/java/com/jnjcomu/edison/EdisonApplication.java
package com.jnjcomu.edison;

import android.app.Application;

import com.jnjcomu.edison.callback.CloudEventListener;
import com.jnjcomu.edison.cloud.LoplatCloudListener;
import com.jnjcomu.edison.storage.Configuration;
import com.loplat.placeengine.Plengi;

import org.androidannotations.annotations.EApplication;

/**
 * @author CodeRi13 <ruto1924@gmail.com>
 * @since 2017-04-12
 */

@EApplication
public class EdisonApplication extends Application {
    private static EdisonApplication instance;

    private Plengi plengi;
    private LoplatCloudListener cloudlistener;

    @Override
    public void onCreate() {
        super.onCreate();

        // NOTE : Please register when login into edison server

        cloudlistener = new LoplatCloudListener();
        plengi = Plengi.getInstance(this);
        plengi.setListener(cloudlistener);

        Configuration conf = new Configuration(this);

        // Please change the "12345" be named user unique code to unique.
        plengi.init(conf.getLoplatId(), conf.getLoplatPw(), "12345");

        instance = this;
    }

    public Plengi getPlengi() {
        return plengi;
    }

    public void setEventListener(CloudEventListener cloudEventListener) {
        if(cloudlistener != null) {
            cloudlistener.setListener(cloudEventListener);
        }
    }

    public void destroyEventListener() {
        cloudlistener = null;
    }

    public static EdisonApplication getInstance() {
        return instance;
    }
}

File: app/src/main/java/com/jnjcomu/edison/EdisonApplication.kt
package com.jnjcomu.edison

import android.app.Application
import com.jnjcomu.edison.addition.configurator
import com.jnjcomu.edison.addition.plengi
import com.jnjcomu.edison.callback.CloudEventListener
import com.jnjcomu.edison.cloud.LoplatCloudListener

/**
 * @author CodeRi13 <ruto1924@gmail.com>
 * @since 2017-04-12
 */

class EdisonApplication : Application() {
    private var cloudListener = LoplatCloudListener(this)

    override fun onCreate() {
        super.onCreate()

        plengi.listener = cloudListener

        /*
         * NOTE : Please register when login into edison server
         * TODO : "uniqueUserId" should be a individual code
         */
        plengi.init(configurator.loplatId, configurator.loplatPw, "12345")
    }

    fun registerEventListener(cloudEventListener: CloudEventListener) {
        cloudListener.setListener(cloudEventListener)
    }

    fun destroyEventListener() {
        cloudListener.setListener(null)
    }
}
